functions/Get-DbaSqlMemoryCondition

```diff
@@ -0,0 +1,119 @@
+            x.value('(//Record/MemoryNode/CommittedMemory)[1]', 'bigint') / 1024 AS [SQLCommittedMemoryMB], 
+            x.value('(//Record/@id)[1]', 'bigint') AS [RecordId], 
+            x.value('(//Record/@type)[1]', 'varchar(30)') AS [Type], 
+            x.value('(//Record/ResourceMonitor/Indicators)[1]', 'bigint') AS [Indicators], 
+            x.value('(//Record/@time)[1]', 'bigint') AS [RecordTime]
+            FROM (SELECT CAST (record as xml) FROM sys.dm_os_ring_buffers 
+            WHERE ring_buffer_type = 'RING_BUFFER_RESOURCE_MONITOR') AS R(x)) a 
+            CROSS JOIN sys.dm_os_sys_info sys
+            ORDER BY a.[RecordTime] ASC"
+    }
+    process {
+        foreach ($instance in $SqlInstance) {
+            try {
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential -MinimumVersion 9
+            } catch {
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            }
+
+            try {
+                $results = $server.Query($sql)
+            } catch {
+                Stop-Function -Message "Issue collecting data" -Target $instance -ErrorRecord $_
+            }
+            foreach ($row in $results) {
+                [PSCustomObject]@{
+                    ComputerName                   = $server.ComputerName
+                    InstanceName                   = $server.ServiceName
+                    SqlInstance                    = $server.DomainInstanceName
+                    Runtime                        = $row.runtime
+                    NotificationTime               = $row.NotificationTime
+                    NotificationType               = $row.NotificationType   
+                    MemoryUtilizationPercent       = $row.MemoryUtilizationPercent
+                    TotalPhysicalMemoryMB          = $row.TotalPhysicalMemoryMB
+                    AvailablePhysicalMemoryMB      = $row.AvailablePhysicalMemoryMB
+                    TotalPageFileMB                = $row.TotalPageFileMB
+                    AvailablePageFileMB            = $row.AvailablePageFileMB
+                    TotalVirtualAddressSpaceMB     = $row.TotalVirtualAddressSpaceMB
+                    AvailableVirtualAddressSpaceMB = $row.AvailableVirtualAddressSpaceMB
+                    NodeId                         = $row.NodeId
+                    SQLReservedMemoryMB            = $row.SQLReservedMemoryMB
+                    SQLCommittedMemoryMB           = $row.SQLCommittedMemoryMB
+                    RecordId                       = $row.RecordId
+                    Type                           = $row.Type
+                    Indicators                     = $row.Indicators
+                    RecordTime                     = $row.RecordTime
+                    CurrentTime                    = $row.CurrentTime
+
+                } | Select-Object
```

Comment:
Removed this part. Not sure if Select-DefaultView should be used.

Reply:
We use that when there is only a subset of properties to output so it will hide anything not defined, or hide what you exclude.

However for this case I think all the data is rather pertinent but @potatoqualitee may adjust that thought when she tries it out.